src/libsystemd/sd-login/sd-login.c

```diff
@@ -929,10 +931,13 @@ _public_ int sd_machine_get_ifindices(const char *machine, int **ret_ifindices)
                 if (ind < 0)
                         return ind == -EINVAL ? -EUCLEAN : ind;
 
-                ifindices[n++] = ind;
+                if (ifindices)
```

Comment:
you probably want to check ret_ifindices here (or you need to initialize ifindices to NULL up top when ret_ifindices is NULL)

Reply:
Good catch.